libparsec/crates/platform_storage/src/native/workspace/sync_pattern.rs

```diff
@@ -0,0 +1,126 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use libparsec_types::{anyhow, Regex};
+use sqlx::{Connection, Row};
+
+use super::PlatformWorkspaceStorage;
+
+impl PlatformWorkspaceStorage {
+    pub async fn set_prevent_sync_pattern(&mut self, pattern: &Regex) -> anyhow::Result<bool> {
+        let pattern = pattern.to_string();
+
+        let mut transaction = self.conn.begin().await?;
+
+        let fully_applied = db_update_prevent_sync_pattern(&mut transaction, &pattern).await?;
+
+        transaction.commit().await?;
+
+        Ok(fully_applied)
+    }
+
+    pub async fn get_prevent_sync_pattern(&mut self) -> anyhow::Result<(Regex, bool)> {
+        db_get_prevent_sync_pattern(&mut self.conn)
+            .await
+            .and_then(|(pattern, fully_applied)| {
+                Regex::from_regex_str(&pattern)
+                    .map(|re| (re, fully_applied))
+                    .map_err(anyhow::Error::from)
+            })
+    }
+
+    pub async fn mark_prevent_sync_pattern_fully_applied(
+        &mut self,
+        pattern: &Regex,
+    ) -> anyhow::Result<bool> {
+        let pattern = pattern.to_string();
+
+        let mut transaction = self.conn.begin().await?;
+
+        let fully_applied =
+            db_set_prevent_sync_pattern_as_fully_applied(&mut transaction, &pattern).await?;
+
+        transaction.commit().await?;
+
+        Ok(fully_applied)
+    }
+}
+
+async fn db_update_prevent_sync_pattern<T, E>(mut trans: T, pattern: &str) -> anyhow::Result<bool>
+where
+    T: AsMut<E>,
+    for<'c> &'c mut E: sqlx::Executor<'c, Database = sqlx::Sqlite>,
+{
+    let row = sqlx::query(concat!(
+        "UPDATE prevent_sync_pattern",
+        " SET pattern = ?1, fully_applied = false",
+        " WHERE _id = 0 AND pattern != ?1",
+        " RETURNING fully_applied",
```

Comment:
the `RETURNING` seems useless here given if a row is returned, then the `fully_applied = false` has been set